src/topology/connected.lean

```diff
@@ -568,3 +771,138 @@ instance totally_separated_space.of_discrete
 ⟨λ a _ b _ h, ⟨{b}ᶜ, {b}, is_open_discrete _, is_open_discrete _, by simpa⟩⟩
 
 end totally_separated
+
+section connected_component_setoid
+
+/-- The setoid on a topological space given by the connected components of it -/
+def connected_component_setoid (α : Type*) [topological_space α] : setoid α :=
+⟨ λ x y, connected_component x = connected_component y,
+  ⟨ λ x, by trivial, λ x y h1, eq.symm h1, λ x y z h1 h2, eq.trans h1 h2 ⟩⟩
+local attribute [instance] connected_component_setoid
+
+lemma component_rel_iff {x y : α} : ⟦x⟧ = ⟦y⟧ ↔ connected_component x = connected_component y :=
+⟨λ h, (quotient.exact h), λ h, quotient.sound h⟩
+
+notation `π₀ ` α :max := quotient (connected_component_setoid α)
```

Comment:
Would it be better to make this a `def`? (I.e., `notation` + `def connected_components :=`)
The notation should probably be localized

Reply:
Like the new commit?

Reply:
Yes, but then `localized`, similar to
```
localized "notation `φ` := nat.totient" in nat
```
Search for `localized` to find more examples. I guess this could go in the `topological_space` locale.

Reply:
Now that I make it as a definition, lint is complaining that "pi0 is missing inhabited instance". I dont really get this issue, do I have to show that there is a topological space of the form pi0 a? Or do I have to show that there exists a function pi0?